src/relay.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 Mateusz Lichota
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <rclcpp/generic_publisher.hpp>
+#include <rclcpp/generic_subscription.hpp>
+
+#include <memory>
+#include <string>
+
+using namespace std::chrono_literals;
```

Comment:
I'd prefer to avoid `using namespace` at the top level, even in `.cpp` files. It's easy enough in this case to say that `discovery_period = std::chrono::milliseconds{100}`